src/flows/context/flow.query.tsx

```diff
@@ -374,20 +374,21 @@ export const FlowQueryProvider: FC = ({children}) => {
           return a
         }, {})
     )
+
     Promise.all(
       map
         .filter(q => !!q.visual)
         .map(stage => {
           return query(stage.visual, stage.scope)
             .then(response => {
-              setStatuses({[stage.id]: RemoteDataState.Done})
               setResult(stage.id, response)
+              setStatuses({[stage.id]: RemoteDataState.Done})
```

Comment:
this change gets rid of a weird flash of the old results would show before showing the new results